crates/holochain_state/src/source_chain.rs

```diff
@@ -972,8 +985,10 @@ fn named_param_seq(base_name: &str, repeat: usize) -> String {
     seq
 }
 
-pub fn lock_for_entry(entry: Option<&Entry>) -> SourceChainResult<Vec<u8>> {
+pub fn chain_lock_subject_for_entry(entry: Option<&Entry>) -> SourceChainResult<Vec<u8>> {
     Ok(match entry {
+        // TODO document that this implies preflight requests must be unique. I.e. if you want to countersign the
+        //      same thing with multiple groups, then you need to use different session times.
```

Comment:
Yes, if you did the "same" session back to back then you'd get different actions and so different session data and a unique entry on the chain.

But the preflight request could be the same. Say you had a preflight request for something like "send 5 holofuel" and you chose to re-use the object because after finishing the first session because you still have plenty of time remaining in the session times. That would be a valid preflight request from Holochain's point of view but would serialize+hash to the same value. So, you should always build a strictly new preflight request.

It's one of those things that could be caught by app validation but I think it's best just to always build a new preflight request and that's what this TODO is to remind me to document.

Reply:
@ThetaSinner maybe i'm imagining things, but i thought the preflight request includes the chain head that everyone has to accept is the state of their counterparties before entering the session

if it does, then you can't reuse requests like that

Reply:
Everyone's chain head goes in their preflight response. So everyone can agree to the preflight same preflight request when they receive it but they will produce their own response at that point. All of these responses combined go into the session data which is what each agent can use for validation of the state of each participant